lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -149,12 +99,10 @@ def generate_secret_yaml(secret_name, secret_type, data)
         },
         "data" => encoded_data,
       }
-      secret.to_yaml
-    end
 
-    def secret_exists?(secret_name)
-      _out, _err, st = @kubectl.run("get", "secret", secret_name)
-      st.success?
+      KubernetesDeploy::Secret.build(
+        namespace: @namespace, context: @context, logger: @logger, definition: secret, statsd_tags: [],
```

Comment:
`statsd_tags` should get the same set as other resources, so either we need to pass that set to this class, or we need to make this class return the definition only and make `DeployTask` build the resource.